openpgp-sdkms/src/lib.rs

```diff
@@ -877,6 +947,166 @@ fn api_curve_from_sequoia_curve(curve: SequoiaCurve) -> Result<ApiCurve> {
     }
 }
 
+fn secret_packet_from_sobject(
+    sobject: Sobject,
+    role: KeyRole,
+) -> Result<Packet> {
+    let time: SystemTime = sobject.created_at.to_datetime().into();
+    let raw_secret = sobject.value.context("secret bits missing in Sobject")?;
+    let raw_public =
+        sobject.pub_key.context("public bits missing in Sobject")?;
+    let is_signer =
+        (sobject.key_ops & KeyOperations::SIGN) == KeyOperations::SIGN;
+    match sobject.obj_type {
+        ObjectType::Ec => match sobject.elliptic_curve {
+            Some(ApiCurve::Ed25519) => {
+                let secret = &raw_secret[16..];
```

Comment:
need some knowledge sharing: why we start from 16?

Reply:
The first bytes contain constant identifiers (algorithm identifier, curve identifier) defined in [RFC8410](https://datatracker.ietf.org/doc/html/rfc8410). That said, I should probably either parse the thing and check those identifiers, or at the very least,  gracefully error if `raw_secret` has less than 16 bytes.

Reply:
- [ ] Check length of `raw_secret`